xwiki-platform-core/xwiki-platform-attachment/xwiki-platform-attachment-ui/src/main/resources/XWiki/AttachmentSelector.xml

```diff
@@ -107,12 +110,23 @@ $xwiki.jsx.use($attachmentPickerDocName)
  * @param $currentValue the currently selected file, used for determining if the box should be highlighted as the current value
  *#
 #macro (attachmentPicker_displayAttachmentBox $attachment $targetDocument $targetAttachDocument, $options $currentValue)
-  #if ($options.displayImage &amp;&amp; $attachment.isImage())
-    #set ($cssClass = 'gallery_image')
+  #set ($hasTemporaryAttachment = "$!services.temporaryAttachments" != '')
+  #set ($canEdit = $xwiki.hasAccessLevel('edit', $xcontext.user, ${targetAttachDocument.fullName}))
+  #set ($isTemporaryAttachment = false)
+  #if(!$hasTemporaryAttachment)
+    #set ($canDeleteAttachment = $canEdit)
   #else
-    #set ($cssClass = '')
+    #set ($isTemporaryAttachment = $services.temporaryAttachments.temporaryAttachmentExists($attachment))
+    #set ($canDeleteAttachment = !$isTemporaryAttachment &amp;&amp; $canEdit)
```

Comment:
Probably need a TODO comment for future improvment: it means that if you're using the macro in edit mode, then the temp attachments you added will probably be saved to the doc. So it would make sense to allow delete the temp attachment, especially since you should not require edit rights for those.

Reply:
https://jira.xwiki.org/browse/XWIKI-20225 created, adding TODO.

Reply:
Done, see https://github.com/xwiki/xwiki-platform/pull/1930/files#diff-e1513599ab698991f6cbba55d38f3f464432ced8d137a668b1f7618c7e747e23R120